tests/functional/horizon_controller_test.go

```diff
@@ -145,4 +146,21 @@ var _ = Describe("Horizon controller", func() {
 			Expect(memcached).NotTo(BeNil())
 		})
 	})
+	When("Route Annotation is supplied", func() {
+		BeforeEach(func() {
+			DeferCleanup(DeleteInstance, CreateSharedMemcached())
+			DeferCleanup(DeleteInstance, CreateHorizonSharedMemcachedWithAnnotation(horizonName))
+			secret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      SecretName,
+					Namespace: namespace,
+				},
+			}
+			Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+		})
+		It("Should have an Annotation on the Route", func() {
+			horizonRoute := k8sClient.List(ctx, &routev1.RouteList{})
+			fmt.Printf("debug: HorizonRoute: %+v", horizonRoute)
+		})
+	})
```

Comment:
Need to use the Route helper from lib-common to test this:
https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/test/openshift_crds/README.md